tests/core/test_addresses_correctly_assigned.py

```diff
@@ -0,0 +1,104 @@
+import pytest
+import time
+
+from django.test import RequestFactory
+
+from pretix_eth.models import WalletAddress, WalletAddressError
+
+hex_addresses = [
+    '0x4257a60613500E8272b0d6151E6CCcAcf79E8Ff4',
+    '0x7D3013fB1Abe40A7bfBAA5E826E0A178F01Ef2F9',
+    '0x4B10a9cbE018B01Ef65B43c622fE0Bd08a8BA160',
+    '0xf5773255D551156182e7F0E3A26bD8d0Bb0022B7',
+    '0xF68b56523CF6ec929953159B67e250a98726349B',
+]
+
+
+@pytest.mark.django_db
+def test_addresses_correctly_assigned(event, provider, get_order_and_payment):
+    provider.settings.set('ETH_RATE', '1.0')
+
+    for address in hex_addresses:
+        WalletAddress.objects.create(event=event, hex_address=address)
+
+    assert WalletAddress.objects.filter(order_payment__isnull=True).count() == len(hex_addresses)
+
+    used_addresses = set()
+
+    for _ in range(len(hex_addresses)):
+        factory = RequestFactory()
+        request = factory.get('/checkout')
```

Comment:
It's generally best to not use hard coded "string" URLS, and instead to use [`django.urls.reverse`](https://docs.djangoproject.com/en/3.1/ref/urlresolvers/#reverse) to lookup the URL so that *if* the URL changes, all of the strings don't also have to be updated.

Reply:
This object is not being used at all. So, I am considering removing it, but not sure what will be the best practice because payment_pending_render() still requires it (but does not use it).

Reply:
If you are going to remove it, do it in a separate PR.  For now, can you open an issue just to track the potential change.